File: RegexMatch.java
import java.util.regex.Pattern;
import java.util.regex.Matcher;

public class RegexMatch {

        public static void main(String[] args) {
                String input = "minhaz 42";

                // match alphanumeric variable = some number
                Pattern p = Pattern.compile("(\\w) (\\d)");
                Matcher m = p.matcher(input);

            if(m.find()) {
                
                String name = m.group(2);
                String number = m.group(3);
                
                System.out.println(name);
                System.out.println(number);
              }
        }

}

File: RegexMatch.cpp
#include <iostream>
#include <string>
#include <regex>

int main ()
{
    std::string line("minhaz 42");
    std::regex pattern("([a-zA-Z]+) ([0-9]+)");
    
    std::smatch matches;
    std::regex_match (line, matches, pattern);

    for (unsigned i=0; i<matches.size(); ++i) {
    std::cout << "[" << matches[i] << "] ";
    }

    std::cout << std::endl;

    return 0;
}
